# BENG202- Bioinformatics II: Bioinformatic Algorithms
This repository contains implementations and projects related to the UCSD- BENG202/CSE282 course: Bioinformatics II: Bioinformatic Algorithms. Each chapter demonstrates deep understanding of various bioinformatic algorithms and potential applications in solving real-world problems.

## Table of Contents
1. Chapters and Problems
2. Programming Language
3. Contributions
4. Academic Honesty & Integrity
5. Purpose of Publication

## Chapters and Problems
### **Chapter 1:** Where in the genome does replication begin? (Replication Origin)
**1A:** Compute the Number of Times a Pattern Appears in a Text

**1B:** Find the Most Frequent Words in a String

**1C:** Find the Reverse Complement of a String

**1D:** Find All Occurrences of a Pattern in a String

**1E:** Find Patterns Forming Clumps in a String

**1F:** Find a Position in a Genome Minimizing the Skew

**1G:** Compute the Hamming Distance Between Two Strings

**1H:** Find All Approximate Occurrences of a Pattern in a String

**1I:** Find the Most Frequent Words with Mismatches in a String

### **Chapter 2:** Which DNA patterns play a role the role of molecular clocks? (Regulatory Motifs)
**2A:** Implement MotifEnumeration

**2B:** Find a Median String

**2C:** Find a Profile-most Probable k-mer in a String

**2F:** Implement RandomizedMotifSearch

**2G:** Implement GibbsSampler

**2H:** Implement DistanceBetweenPatternAndStrings

### **Chapter 3:** How do we assemble genomes? (Assembly)
**3C:** Construct the Overlap Graph of a Collection of k-mers

**3D:** Construct the De Bruijn Graph of a String

**3E:** Construct the De Bruijn Graph of a Collection of k-mers

**3F:** Find an Eulerian Cycle in a Graph

**3G:** Find an Eulerian Path in a Graph

**3H:** Reconstruct a String from its k-mer Composition

**3I:** Find a k-Universal Circular String

**3J:** Reconstruct a String from its Paired Composition
### **Chapter 4:** How do we compare biological sequences? (Alignment)
**5B:** Find the Length of a Longest Path in a Manhattan-like Grid

**5C:** Find a Longest Common Subsequence of Two Strings

**5D:** Find the Longest Path in a DAG

**5E:** Find a Highest-Scoring Alignment of Two Strings

**5F:** Find a Highest-Scoring Local Alignment of Two Strings

**5G:** Compute the Edit Distance Between Two Strings

**5H:** Find a Highest-Scoring Fitting Alignment of Two Strings

### **Chapter 6:** Are there fragile regions in the human genome? (Rearrangements)
**6A:** Implement GreedySorting to Sort a Permutation by Reversals

**6B:** Compute the Number of Breakpoints in a Permutation

**6E:** Find All Shared k-mers of a Pair of Strings

**6F:** Implement ChromosomeToCycle

**6G:** Implement CycleToChromosome

**6H:** Implement ColoredEdges

## Programming Language
All code written in Python. 

## Contributions
This repository is based on coursework and is not open to contributions. However, feedback and suggestions are always welcome.

## Academic Honesty & Integrity
**Attention Future Students:** If you are currently enrolled in BENG202 or a similar course, I strongly urge you not to copy, replicate, or reference this code directly. Academic dishonesty will not only jeopardize your grade but will also hinder your understanding and long-term success in the field of Bioinformatics.

## Purpose of Publication
This repository has been made public exclusively for my interviewing requirements, allowing potential employers, mentors, or colleagues to review and understand the depth of my coding skills and grasp of computer science concepts. If you are here for reasons related to hiring or professional networking, please feel free to explore the code and reach out to me with any questions or feedback.
