# BENG202- Bioinformatics II: Bioinformatic Algorithms
This repository contains implementations and projects related to the UCSD- BENG202/CSE282 course: Bioinformatics II: Bioinformatic Algorithms. Each chapter demonstrates deep understanding of various bioinformatic algorithms and potential applications in solving real-world problems.

## Table of Contents
1. Chapters
2. Programming Language
3. Contributions
4. Academic Honesty & Integrity
5. Purpose of Publication

## Chapters
**Chapter 1:** Where in the genome does replication begin? (Replication Origin)

**Chapter 2:** Which DNA patterns play a role the role of molecular clocks? (Regulatory Motifs)

**Chapter 3:** How do we assemble genomes? (Assembly)

**Chapter 4:** How do we compare biological sequences? (Alignment, Part 1)

**Chapter 5:** How do we align biological sequences? (Alignment, Part 2)

**Chapter 6:** Are there fragile regions in the human genome? (Rearrangements)

**Chapter 7:** How do we locate disease-causing mutations? (Detecting Mutations, Part 1)

**Chapter 9:** How do we locate disease-causing mutations? (Detecting Mutations, Part 2)

## Programming Language
All code written in Python. 

## Contributions
This repository is based on coursework and is not open to contributions. However, feedback and suggestions are always welcome.

## Academic Honesty & Integrity
**Attention Future Students:** If you are currently enrolled in BENG202 or a similar course, I strongly urge you not to copy, replicate, or reference this code directly. Academic dishonesty will not only jeopardize your grade but will also hinder your understanding and long-term success in the field of Bioinformatics.

## Purpose of Publication
This repository has been made public exclusively for my interviewing requirements, allowing potential employers, mentors, or colleagues to review and understand the depth of my coding skills and grasp of computer science concepts. If you are here for reasons related to hiring or professional networking, please feel free to explore the code and reach out to me with any questions or feedback.
